dotnet/src/dotnetcore/GxClasses.Web/Middleware/GXRouting.cs

```diff
@@ -214,21 +214,24 @@ public Task ProcessRestRequest(HttpContext context)
 							}
 							string controllerPath = path.ToLower().Split(actualPath).Last<string>();
 							controllerWithParms = controllerPath.Split(QUESTIONMARK).First<string>();
-					
+
 						}
 					}
 					else
 					{
-						if (path.Contains(oauthRoute) && (AzureDeploy.GAM == "true")) 
-							return (RouteHttpService(context));
+						if (path.Contains(oauthRoute) && (AzureDeploy.GAM == "true"))
+						{
+							await (RouteHttpService(context));
+							return;
+						}
 						controllerWithParms = GetGxRouteValue(path);
 						GXLogging.Debug(log, $"Running Azure functions. ControllerWithParms :{controllerWithParms} path:{path}");
 					}
-				
+
 					List<ControllerInfo> controllers = GetRouteController(servicesPathUrl, servicesValidPath, servicesMap, servicesMapData, actualPath, context.Request.Method, controllerWithParms);
 					GxRestWrapper controller = null;
 					ControllerInfo controllerInfo = controllers.FirstOrDefault(c => (controller = GetController(context, c)) != null);
-					
+
 
 					if (controller != null)
```

Comment:
There are several assignments to the `result` local variable inside this code path. It seems previously this method would return the resulting task. In order to keep the same behavior, now that the method is async, it should await for the task, instead of assigning it to a local variable. I'd suggest removing the `result` variable, and await on each place where it was previously assigned.